### DISCUSSION TITLE
How to get 'HasUniqueRoleAssignments' of item when using 'LoadListDataAsStream' approach

### DISCUSSION THREAD

**Discussion by bajce** [pnp/pnpcore#1138] [Q&A]
In order to avoid exception when there are more then 5000 items in Folder/List, I'm using `LoadListDataAsStream` approach as suggested in [#1044 ](https://github.com/pnp/pnpcore/issues/1044). 
Unfortunately `HasUniqueRoleAssignments` property of the loaded items is not set/available.
Due my research I found that when one Item is shared, there are additional `FieldValue` properties created like `SharedWithUsers` and `SharedWithDetails` but results are not equivalent with the `HasUniqueRoleAssignments`. 
Is there any way to load the `HasUniqueRoleAssignments` while loading the items by `LoadListDataAsStream`  and without extra server round-trips or is there any FieldValue which indicates that this item has unique permissions?

**Comment by jansenbe** [comment]
@bajce : did you check the raw JSON that gets returned from the `LoadListDataAsStream` call when you load the `HasUniqueRoleAssignments` property? Maybe it's returned but not correctly processed? If it's not returned then I do not see another way. There's no other field data that correctly defines this, the fact that this is costly server side operation is also the reason why you can't filter on `HasUniqueRoleAssignments` in the other list item retrieval approaches.

**Comment by bajce** [comment]
@jansenbe: Raw JSON returned from the SharePoint API when calling `LoadListDataAsStream` do not include `HasUniqueRoleAssignments` (even do is provided in `<ViewFields>` tag as `<FieldRef Name='SharedWithDetails'/>`. I also tried couple of more approaches to additionally select `HasUniqueRoleAssignments` as request query parameter but id didn't change anything. 
Im wondering if there is no option to bulk load `HasUniqueRoleAssignments` for `ListItems`, how does the `LoadItemsByCamlQuery` does that? Or is the property loaded explicit for each ListItem (in one batch request)?

**Reply by jansenbe** [reply]
You can indeed load this property per list item and batch them to save on roundtrip. Not optimal but can work
